sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -141,7 +146,8 @@ case class GpuFilterExec(condition: Expression, child: SparkPlan)
   /**
    * Potentially a lot is removed.
    */
-  override def coalesceAfter: Boolean = true
+  override def coalesceAfter: Boolean =
+    !child.expressions.exists(expr => expr.dataType.isInstanceOf[MapType])
```

Comment:
We need to fix CoalesceBatch not this.

Reply:
I have added a new iterator as a combination of code from https://github.com/NVIDIA/spark-rapids/pull/722/ and https://github.com/NVIDIA/spark-rapids/pull/833 plus a couple api fixups and removed filter exec changes for it. I checked that for target query with filter on maps it works but given my newbie status with recent coalesce changes, it would be nice to get you to correct anything I might have missed. @revans2